packages/aleph-holders/src/api/types.ts

```diff
@@ -27,17 +27,37 @@ export const ERC20TransferEvent = new GraphQLObjectType({
   },
 })
 
-export const ERC20TransferEventList = new GraphQLList(ERC20TransferEvent)
+export const TransferEventList = new GraphQLList(TransferEvent)
 
-export const ERC20Balance = new GraphQLObjectType({
-  name: 'ERC20Balance',
+export const Balance = new GraphQLObjectType({
+  name: 'Balance',
```

Comment:
balance query also can return solana balances from snapshot db

Reply:
I see now where the problem in [this discussion](https://github.com/aleph-im/aleph-indexer-library/pull/19#discussion_r1565587408) lies.

What you can do is make `Balance` a union of two types, `ERC20Balance` and `SolanaBalance`:
```ts
const erc20BalanceFields = {
  account: { type: new GraphQLNonNull(GraphQLString) },
  balance: { type: new GraphQLNonNull(GraphQLBigNumber) },
  balanceNum: { type: GraphQLFloat },
}

const solanaBalanceFields = {
  ...erc20BalanceFields,
  tokenAccount: { type: new GraphQLNonNull(GraphQLString) },
}

export const ERC20Balance = new GraphQLObjectType({
  name: 'ERC20Balance',
  fields: erc20BalanceFields,
})

export const SolanaBalance = new GraphQLObjectType({
  name: 'SolanaBalance',
  fields: solanaBalanceFields,
})

export const Balance = new GraphQLUnionType({
  name: 'Balance',
  types: [ ERC20Balance, SolanaBalance],
});
```
This will allow you to keep all the relevant info for Solana-related balances, while not having to deal with undefined values for ERC20Balances. This style also comes in handy later, when you might want to filter balances based on which type they are.